packages/mobx-state-tree/src/core/mst-operations.ts

```diff
@@ -210,7 +210,7 @@ export function recordPatches(subject: IAnyStateTreeNode): IPatchRecorder {
             applyPatch(target || subject, recorder.patches)
         },
         undo(target?: IAnyStateTreeNode) {
-            applyPatch(target || subject, recorder.inversePatches.slice().reverse())
+            applyPatch(target || subject, recorder.inversePatches.reverse())
```

Comment:
The `.slice` was here as `reverse` tends to update in place

Reply:
that was done because it uses recorder.inversePatches, which is:
```
get inversePatches() {
            return this.rawPatches.map(([_, b]) => b)
        },
```
and since map does a copy it means the slice makes a copy of a copy

Reply:
I think ill change it so it doesn't need to do all those transformations

Reply:
Improved in this commit: https://github.com/mobxjs/mobx-state-tree/pull/1196/commits/0df3a2ff5145bfc278ae7dfae4b248f423950ba6